powersimdata/tests/mock_scenario.py

```diff
@@ -34,3 +38,26 @@ def __init__(self, grid_attrs=None, **kwargs):
     def __class__(self):
         """If anyone asks, I'm a Scenario object!"""
         return Scenario
+
+    def __dir__(self):
+        return sorted(super().__dir__() + list(self.state.exported_methods))
+
+    def __getattr__(self, name):
+        if name in self.state.exported_methods:
+            return getattr(self.state, name)
+        elif hasattr(self.state, "__getattr__"):
+            return self.state.__getattr__(name)
+        else:
+            raise AttributeError(
+                f"Scenario object in {self.state.name} state "
+                f"has no attribute {name}"
+            )
+
+    def __setattr__(self, name, value):
+        if name in self._setattr_allowlist:
+            super().__setattr__(name, value)
+        elif name in self.state.exported_methods:
+            raise AttributeError(
+                f"{name} is exported from Scenario.state, edit it there if necessary"
+            )
+        super().__setattr__(name, value)
```

Comment:
I know we did the same thing as in `scenario.py` here. But I'm still wondering given we do `super().__setattr__(name, value)` by default, why we need `if name in self._setattr_allowlist`?

Reply:
If we don't, we can't set the `state` attribute during init.

Reply:
So if we do 
```python
    def __setattr__(self, name, value):
        if name in self.state.exported_methods:
            raise AttributeError(
                f"{name} is exported from Scenario.state, edit it there if necessary"
        super().__setattr__(name, value)
```
We can't set the state attribute with `self.state = MockAnalyze(grid_attrs, **kwargs)` in `init`?

Reply:
Setting `self.state` in init uses `__setattr__`, which will get to `if name in self.state.exported_methods:` and raise an AttributeError since `self.state` doesn't exist yet.

Reply:
Got it. Make sense!